workflow/executor/executor.go

```diff
@@ -117,6 +117,10 @@ func (we *WorkflowExecutor) LoadArtifacts() error {
 		log.Infof("Downloading artifact: %s", art.Name)
 		artDriver, err := we.InitDriver(art)
 		if err != nil {
+			if art.Optional {
```

Comment:
I think we need to be specific about the error. With this logic, we can't distinguish a driver initialization errors vs. an error because the artifact location is not supplied. The former should fail the workflow. The latter should not.

Alternatively, we can maybe avoid calling InitDriver entirely and simply continue the loop, if we know that the artifact has no location (or has an empty location), and the artifact was optional

Reply:
done